### PR TITLE
Documenting names of `state_vars` variables; docstrings

### PR DESCRIPTION
**Pull request by mike-khor**
Not a good solution, just a bandaid. Adding a list of AUXILIARY_CONDITIONS for referencing the names of variables returned by `state_vars`. This list will update over the evolution of `growth_model`.

### CODE REVIEW

**Review by amalhotra40 — APPROVED** [review]
lgtm, just fyi its running into mypy troubles @mike-khor

File: insilicho/growth_model.py [hunk 1]
```diff
@@ -30,13 +30,47 @@ def exponential_dependence_around_optima(
...
+        state (np.ndarray): Current states (e.g. Xv, Cglc). Must follow order specified
+            in parameters.InitialConditions.
+        params (parameters.InputParameters): Parameters of GrowCHO model.
+        feed_fn (typing.Optional[FeedFunctionType], optional): Callable describing feed
+            profile. Defaults to None.
+        temp_fn (typing.Optional[TempFunctionType], optional): Callable describing temp
+            profile. Defaults to None.
+
+    Raises:
+        IOError: Raised if `feed_fn` or `temp_fn` are not provided.
```
Comment (svh-cb) [inline insilicho/growth_model.py:68]: I don't think this should be an IOError (not something added in this PR - I realize)

File: insilicho/growth_model.py [hunk 2]
```diff
@@ -30,13 +30,47 @@ def exponential_dependence_around_optima(
     return np.exp(-((x - optima) ** 2.0) / spread**2.0)
 
 
+AUXILIARY_CONDITIONS = [
```
Comment (svh-cb) [inline insilicho/growth_model.py:33]: What's this for? It's new, but looks unused?

File: insilicho/growth_model.py (hunk 1 above)
Comment (mike-khor) [reply insilicho/growth_model.py:68]: The IOError is invoked in L79, but yeah reading it up it's more related to I/O of files. we could switch to ValueError perhaps?

Comment (svh-cb) [reply insilicho/growth_model.py:68]: ValueError sounds better!

File: insilicho/growth_model.py (hunk 2 above)
Comment (mike-khor) [reply insilicho/growth_model.py:33]: it's not used in insiliCHO (yet) but I have a need to use it in `cho_doe` (I made some modifications there). not entirely sure how to tie `AUXILIARY_CONDITIONS` into rest of insiliCHO code. Maybe I could modify `runGrowCHO._full_result.state_vars` to have a dict version of state_vars?

Comment (svh-cb) [reply insilicho/growth_model.py:33]: hm I'd say the simplest is, if you only need it in cho_doe, just keep it there for now?

**Review by svh-cb — APPROVED** [review]
not sure the auxiliary conditions should be here as opposed to cho_doe, but no objection to merging as is

File: insilicho/growth_model.py (hunk 2 above)
Comment (mike-khor) [reply insilicho/growth_model.py:33]: fair point. i'll make this a docstring thing only then

**Review by svh-cb — APPROVED** [review]

**Pull request merged**